enterprise/views.py

```diff
@@ -498,35 +499,68 @@ def set_final_prices(self, modes, request):
             result.append(mode)
         return result
 
-    def get_base_details(self, enterprise_uuid, course_run_id):
+    def get_base_details(self, request, enterprise_uuid, course_run_id):
         """
         Retrieve fundamental details used by both POST and GET versions of this view.
 
         Specifically, take an EnterpriseCustomer UUID and a course run ID, and transform those
         into an actual EnterpriseCustomer, a set of details about the course, and a list
         of the available course modes for that course run.
         """
+        enterprise_catalog_uuid = request.GET.get('catalog')
         try:
             enterprise_customer = get_enterprise_customer_or_404(enterprise_uuid)
-            course, course_run = \
-                CourseCatalogApiServiceClient(enterprise_customer.site).get_course_and_course_run(course_run_id)
         except ImproperlyConfigured:
             raise Http404
 
+        enterprise_catalog = None
+        if enterprise_catalog_uuid:
+            try:
+                enterprise_catalog = enterprise_customer.enterprise_customer_catalogs.get(
+                    uuid=enterprise_catalog_uuid
+                )
+            except EnterpriseCustomerCatalog.DoesNotExist:
+                add_generic_info_message_for_error(request)
+
+        if enterprise_catalog:
+            course, course_run = enterprise_catalog.get_course_and_course_run(course_run_id)
+        else:
+            try:
+                course, course_run = CourseCatalogApiServiceClient(
+                    enterprise_customer.site
+                ).get_course_and_course_run(course_run_id)
+            except ImproperlyConfigured:
+                raise Http404
```

Comment:
If the `catalog` query parameter was provided and we can't get the course and course_run data for any reason, whether that be because there was a problem retrieving the `EnterpriseCustomerCatalog` for the UUID provided or because the course run does not exist in the specified catalog, we should should display the landing page with an error message, the course details (if we are able to retrieve them), and NO mode options or continue button. It seem like we should be able to simplify the code here looking at things from that approach.

Reply:
Replacing `raise Http404` with `add_generic_info_message_for_error(request)` remaining two places is causing multiple tests to fail. We should create a separate story for remaining refactoring.

Reply:
Sounds good. Can you create those tickets and tag me?

Reply:
I think the existing https://openedx.atlassian.net/browse/ENT-688 would be enough?

Reply:
@douglashall @UmanShahzad I have created a separate ticket [ENT-717](https://openedx.atlassian.net/browse/ENT-717) just in case.